action/protocol/staking/handlers.go

```diff
@@ -186,25 +203,28 @@ func (p *Protocol) handleWithdrawStake(ctx context.Context, act *action.Withdraw
 	}
 
 	log := p.createLog(ctx, HandleWithdrawStake, nil, actionCtx.Caller, nil)
-	return p.settleAction(ctx, sm, gasFee, log)
+	return p.settleAction(ctx, sm, uint64(iotextypes.ReceiptStatus_Success), gasFee, log)
 }
 
 func (p *Protocol) handleChangeCandidate(ctx context.Context, act *action.ChangeCandidate, sm protocol.StateManager) (*action.Receipt, error) {
 	actionCtx := protocol.MustGetActionCtx(ctx)
 
-	candidate := p.inMemCandidates.GetByName(act.Candidate())
-	if candidate == nil {
-		return nil, errors.Wrap(ErrInvalidCanName, "cannot find candidate in candidate center")
+	_, gasFee, fetchErr := fetchCaller(ctx, sm, big.NewInt(0))
+	if fetchErr != nil {
+		log.L().Debug("Error when fetching caller", zap.Error(fetchErr.err))
+		return p.settleAction(ctx, sm, uint64(fetchErr.failureStatus), gasFee)
 	}
 
-	_, gasFee, err := fetchCaller(ctx, sm, big.NewInt(0))
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to fetch caller")
+	candidate := p.inMemCandidates.GetByName(act.Candidate())
+	if candidate == nil {
+		log.L().Debug("Error when finding candidate in candidate center", zap.Error(ErrInvalidCanName))
+		return p.settleAction(ctx, sm, uint64(iotextypes.ReceiptStatus_ErrCandidateNotExist), gasFee)
 	}
 
-	bucket, err := p.fetchBucket(ctx, sm, act.BucketIndex(), true, false)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to fetch bucket")
+	bucket, fetchErr := p.fetchBucket(ctx, sm, act.BucketIndex(), true, false)
+	if fetchErr != nil {
+		log.L().Debug("Error when fetching bucket", zap.Error(fetchErr.err))
+		return p.settleAction(ctx, sm, uint64(fetchErr.failureStatus), gasFee)
 	}
 
 	prevCandidate := p.inMemCandidates.GetByOwner(bucket.Candidate)
```

Comment:
we should also settle this error?

Reply:
I didn't settle this error in receipt because the candidate is already associated with a bucket. If the candidate center cannot find this candidate, it is not the action caller's fault, but something wrong with the system.